### DISCUSSION TITLE
Backup fills host disk with a docker volume

### DISCUSSION THREAD

**Discussion by mikeymop** [garethgeorge/backrest#411] [Q&A]
Hello, I am struggling to achieve a successful initial backup to a restic repo I created on the internet.

I am using the container image with the following volume configuration:

```
/userdata # mounted my NAS mergerfs pool
/data     # arbitrary directory on host
/config   # arbitrary directory on host
/cache    # arbitrary directory on host
/resticTemp # $HOME/resticTemp on host
```

The backup starts and runs for several hours but eventually terminates with the following error:
```
Fatal: unable to save snapshot: Write: write /tmp/restic-temp-pack-1690927867: no space left on device
```

Initially I attributed this to me not ignoring the directories I mounted for cache and data so I added these ignore patterns to the backup:
```
/userdata/backrest/cache/**/*
/userdata/torrent/downloads/**/*
```

This produces the following restic command:

```
/bin/restic-0.16.4 backup --json /userdata -osftp.args= -oBatchMode=yes --exclude-caches --tagplan:Monthly-Backup--tagcreated-by:foo --exclude/userdata/backrest/cache/**/* --exclude/userdata/downloads/**/*--no-cache
```

This resulted in a full disk. I have 684gb oh data on the drive I wish to backup, and 350gb of free space on the host's disk.

I suspected this might be the tmpfs filling so I've added the following env vars to the container:
```
TMPDIR=/resticTemp
```

Which is associated with the `$HOME/resticTemp:/resticTemp` volume mount I've added in an attempt to mitigate this issue.

Is consuming nearly 350gb of cache in order to perform an initial backup a reasonable amount?
Am I misconfiguring backrest, or the plan, in a specific way causing the cache to balloon in size?

Any help would be appreciated.

**Comment by garethgeorge** [comment]
Hmm, off the top of my head 

 * TMPDIR is used for creating temporary files during upload etc iiuc.
 * XDG_CACHE_HOME is another temporary location that restic uses to cache repo files w/many packs. 

You might try making sure you are setting both directories. I wonder if it's actually your cache that's filling the tmpfs? This may also be a question worth asking about on the restic forums if you are actually seeing 350gb of writes.

**Comment by devdecrux** [comment]
@mikeymop, @garethgeorge hi, did you manage to solve this issue? I am struggling with exactly the same thing. I am running backrest on an lxc container with 8GB of HDD but in the same time I've mounted my nfs share where I am trying to set the TMPDIR to it through the repository settings, but without any luck.

TMPDIR=/mnt/nfs-share/restic/tmp
RESTIC_CACHE_DIR=/mnt/nfs-share/restic/cache
XDG_CACHE_HOME=/mnt/nfs-share/restic/cache

Those are the environment variables which I've set throught the REPOSITORY settings in backrest. Yet, I see how in my lxc container under /s3.eu-central-id.backblazeb2.com/bucket_name/data is being filled up with the data I am trying to backup... am I missing some env var?
